seqr/views/utils/file_utils.py

```diff
@@ -7,6 +9,7 @@
 import tempfile
 import openpyxl as xl
 
+from django.core.files.uploadedfile import InMemoryUploadedFile
```

Comment:
This seems like something that is only added for testing purposes. I STRONGLY disagree with having conditional logic designed for that purpose, it makes the real code more clunky and it means we aren't actually testing what we want to be testing. If this is an issue with how the mock file isbeing passed, try using `BytesIO` as your test input instead of a utf-8 encoded string

Reply:
I agree that using isinstance(InMemoryUploadedFile) is not elegant. The issue is parse_file() can have an either binary (which is a InMemoryUploadedFile) or text stream inputs. I finally find a way to convert the binary input into a text input by using io.TextIOWrapper.

Reply:
But why are we supporting binary files at all? Is there any "real" use case with binary, or is it testing only? If testing only, you should change how we are testing, not the real code

Reply:
It is not for testing only. I thought all the input data must be in text format. That was true if we don't support `.xls` files.
Since we need to support both text and binary data, the text decoding has to be done inside the parse_file() function. I will figure out how to do it and ask you review later.